beta-src/src/models/BoardClass.ts

```diff
@@ -0,0 +1,261 @@
+
+          currUnit.setTerritory(currTerritory);
+
+          this.units.push(currUnit);
+        }
+      }
+
+      if (currTerritory.coast === CoastType.Parent) {
+        coastParents.push(currTerritory);
+      }
+
+      if (currTerritory.coast === CoastType.Child) {
+        coastChildren.push(currTerritory);
+      }
+
+      if (currTerritory.coastParentID === currTerritory.id) {
+        currTerritory.coastParent = currTerritory;
+      }
+
+      return currTerritory;
+    });
+
+    coastChildren.forEach((cc) => {
+      const parent = coastParents.find((cp) => cp.id === cc.coastParentID);
+      if (parent) {
+        cc.setCoastParent(parent);
+      }
+    });
+
+    /**
+     * load and initialize convoyGroup.
+     * convoyGroup has
+     * fleets that are adjacent
+     * coasts that are adjacent above fleets
+     * armies that are placed in above coasts
+     */
+    if (context.phase === GamePhaseType.Diplomacy) {
+      this.units.forEach((u) => {
+        if (
+          u.type === UnitType.Fleet &&
+          u.Territory.type === TerritoryType.Sea
+        ) {
+          const newConvoyGroup = new ConvoyGroupClass(this);
+
+          /**
+           * first load fleets, and coasts with armies within them.
+           */
+          newConvoyGroup.loadFleet(u);
+          newConvoyGroup.loadCoasts();
+          this.convoyGroups.push(newConvoyGroup);
+        }
+      });
+
+      this.convoyGroups.forEach((cg) => cg.linkGroups());
+      this.convoyGroups.forEach((cg) => cg.linkGroups());
+    }
+
+    if (context.phase === GamePhaseType.Build) {
+      this.territories.forEach((t) => {
+        if (
+          t.coastParent.supply &&
+          t.coastParent.countryID === context.countryID &&
+          t.coastParent.ownerCountryID === context.countryID &&
+          !t.coastParent.Unit
+        ) {
+          this.supplyCenters.push(t);
+        }
+      });
+    }
+  }
+
+  /**
+   *
+   * @param unitID
+   * @returns {UnitClass}
+   */
+  findUnitByID(unitID: string): UnitClass | undefined {
+    return this.units.find((unit) => unit.id === unitID);
+  }
+
+  /**
+   *
+   * @param territoryID
+   * @returns {TerritoryClass}
+   */
+  findTerritoryByID(territoryID: string): TerritoryClass | undefined {
+    return this.territories.find((t) => t.id === territoryID);
+  }
+
+  /**
+   * finds adjacent territories
+   * @param territory
+   * @returns {TerritoryClass[]}
+   */
+  getBorderTerritories(territory: TerritoryClass): TerritoryClass[] | [] {
+    const borderIDs = territory.Borders.map((b) => b.id);
+
+    return this.territories.filter((t) => borderIDs.includes(`${t.id}`));
+  }
+
+  /**
+   * find units on border territories
+   * @param territory
+   * @returns {UnitClass[]}
+   */
+  getBorderUnits(territory: TerritoryClass): UnitClass[] {
+    const borderTerritories = this.getBorderTerritories(territory.coastParent);
+
+    return borderTerritories.map((bt) => bt.Unit);
```

Comment:
I think this can be a single line, don't see the use of a variable here. 

`return this.getBorderTerritories(territory.coastParent).map((bt) => bt.Unit);`